go/ql/lib/semmle/go/frameworks/Logrus.qll

```diff
@@ -101,12 +101,37 @@ module Logrus {
 
   /**
    * Holds if `node` is the first argument to a call to the `SetFormatter` function or if `node`
-   * is the value being assigned to the `Formatter` property of a `Logger` object.
+   * is the value being assigned to the `Formatter` property of a `Logger` object. The `expr`
+   * is the corresponding expression (the function call or the assignment).
    */
-  private predicate isFormatter(DataFlow::Node node) {
-    node = any(SetFormatterFunction f).getACall().getArgument(0)
+  private predicate isFormatter(DataFlow::Node node, Expr expr) {
+    exists(DataFlow::CallNode call |
+      call = any(SetFormatterFunction f).getACall() and
+      node = call.getArgument(0) and
+      expr = call.asExpr()
+    )
     or
-    node.asExpr() = any(SetFormatterAssignment stmt).getRhs()
+    expr = any(SetFormatterAssignment stmt).getRhs() and
+    node.asExpr() = expr
+  }
+
+  /**
+   * Holds if `expr` is conditional within its enclosing function and there are no other formatter
+   * assignments in the same function.
+   */
+  private predicate isSoleConditional(Expr expr) {
+    exists(FuncDef func |
+      // find the enclosing function
+      func = expr.getEnclosingFunction() and
```

Comment:
Not all exprs have an enclosing function. There can be declarations at the top level. I think we might be okay in this case though.

Reply:
It doesn't look like `if`-statements can appear at the top level, so I suppose this should be fine here.